library/alloc/src/sync.rs

```diff
@@ -1386,17 +1392,14 @@ impl<T: Clone> Arc<T> {
 
             // Materialize our own implicit weak pointer, so that it can clean
             // up the ArcInner as needed.
-            let weak = Weak { ptr: this.ptr };
+            let _weak = Weak { ptr: this.ptr };
 
-            // mark the data itself as already deallocated
+            // Can just steal the data, all that's left is Weaks
+            let mut arc = Self::new_uninit();
             unsafe {
-                // there is no data race in the implicit write caused by `read`
-                // here (due to zeroing) because data is no longer accessed by
-                // other threads (due to there being no more strong refs at this
-                // point).
```

Comment:
I have to admit, I don't really understand what this was saying about zeroing. Maybe that's outdated from the days of drop flags and such? The comment came from commit d77c4b0fa63d420b2f995e37ee07402f9a243361.